docs_src/design/adr/0027-URIs for Files.md

```diff
@@ -0,0 +1,113 @@
+- Lenny Goodell (Intel)
+
+## Change Log
+- [pending ](https://github.com/edgexfoundry/edgex-docs/pull/949) (2023-02-01)
+
+## Referenced Use Case(s)
+- [URIs for Files UCR](https://docs.edgexfoundry.org/3.0/design/ucr/URIs-for-Files/)
+
+## Context
+Currently, in the Levski and earlier releases services can only load configuration, units of measurements, device profiles, device definitions, provision watches, etc. from the local file system. As outlined in the reference UCR, there is a need to be able to load files from a remote locations using URIs to specify the locations. 
+
+## Proposed Design
+This ADR proposes a new helper function for loading files be added to `go-mod-bootstrap`. This function will provide the logic for loading a file either from local file system (as is today) or from a remote location. As stated in the UCR, only **HTTP** and **HTTPS** URIs will be supported. For **HTTPS**, certificate validation will be performed using the system's built-in trust anchors. The docker images for all services will have the CA certs installed as is done here in App Service Configurable's [Dockerfile](https://github.com/edgexfoundry/app-service-configurable/blob/v2.3.0/Dockerfile#L46).
+
+### Authentication 
+
+#### basic-auth in URI
+
+While not recommended, users will be able to specify **basic-auth** (`<username>:<password>@`) in the URI in plain text. While this is ok network wise when using HTTPS, it isn't good practice to have these credentials specified in configuration or other service files where the URI is specified.
+
+!!! example - "Example plain text `basic-auth` in URI located in configuration"
+    ```toml
+    [UoM]
+    UoMFile = "https://myuser:mypassword@example.com/uom.yaml"
+    ```
+
+#### Secure Credentials
+
+In order to provide a secure way for users to specify credentials, the `edgexSecretName` query parameter can be specified on the URI. This parameter specifies a Secret Name from the service's Secret Store where the credentials reside and will be processed by the new helper function.
+
+!!! example - "Example URI with `edgexSecretName` query parameter"
+    ```toml
+    [UoM]
+    UoMFile = "https://example.com/uom.yaml?edgexSecretName=mySecretName"
+    ```    
+
+The type of authentication as well as the credentials will be contained in the secret data specified by the Secret Name. Three types of authentication will be supported, which are `usernamepassword`, `apikey` and `bearertoken`. 
+
+- When `usernamepassword` is specified as the type in the secret data, the **basic-auth** `username:password@` will be inserted into the URI using the **username** and **password** found in the secret data.
+
+    !!! example - "Example secret data - `usernamepassword`"
+        ```
+        type=usernamepassword
+        username=myuser
+        password=mypassword
+        ```
+    !!! example  - "Example resulting URI with `basic-auth`"
+        ```
+        https://myuser:mypassword@example.com/uom.yaml
+        ```

```

Comment:
How about

```suggestion
        ```
        type=customheader
        headername=Authorization
        headercontents=Basic bXl1c2VyOm15cGFzc3dvcmQ=
        ```
        For a request header set as:
        ```
        GET https://example.com/uom.yaml HTTP/1.1
        Authorization: Basic bXl1c2VyOm15cGFzc3dvcmQ=
        ```
```

Reply:
OK, @bnevis-i , what you are proposing is a more general form allowing for more then `Bearer`?

Reply:
@lenny-intel Right.  All of the examples have a static http header and a static value -- just implement that and save some code.

Reply:
replace `token` with you `customheader` and remove `apikey`